modules/standard/Random.chpl

```diff
@@ -986,9 +991,50 @@ module Random {
       proc choice(arr: [], size:?sizeType=none, replace=true, prob:?probType=none)
         throws
       {
-        return _choice(this, arr, size=size, replace=replace, prob=prob);
+        var A = arr.reindex(1..arr.size);
+        const low = arr.domain.alignedLow,
+              stride = abs(arr.domain.stride);
+        var X = A.domain;
+        return _choice(this, X, stride=stride, low=low, size=size, replace=replace, prob=prob);
       }
 
+      /*
+     Returns a random sample from a given 1-D range, ``rng``.
+
+     :arg rng: a 1-D range with values that will be sampled from.
+     :arg size: An optional integral value specifying the number of elements to
+                choose, or a domain specifying the dimensions of the
+                sampled array to be filled, otherwise a single element will be
+                chosen.
+     :arg replace: an optional ``bool`` specifying whether or not to sample with
+                   replacement, i.e. elements will only be chosen up to one
+                   time when ``replace=false``.
+     :arg prob: an optional 1-D array that contains probabilities of choosing
+                each element of ``rng``, otherwise elements will be chosen over
+                a uniform distribution. ``prob`` must have integral or real
+                element type, with no negative values and at least one non-zero
+                value. The domain must be equal to that of ``rng``.
+
+     :return: An element chosen from ``rng`` if ``size == 1``, or an array of
+              element chosen from ``rng`` if ``size > 1`` or ``size`` is a
+              domain.
+
+     :throws IllegalArgumentError: if ``rng.size == 0``,
+                                   if ``rng`` contains a negative value,
+                                   if ``rng`` has no non-zero values.,
+                                   if ``rng(domain) != prob.domain``,
+                                   if ``size < 1 || size.size < 1``,
+                                   if ``replace=false`` and ``size > rng.size || size.size > rng.size``
+     */
+     /*
+      proc choice(rng: range(stridable=true), size:?sizeType=none, replace=true, prob:?probType=none)
+        throws
+      {
+        var A: domain(1,stridable=true) = {rng};
```

Comment:
I have commented this overload for now, it is for strided ranges. The ranges are converted into array and then it is passed to the array overload.

Reply:
What is the intention with this overload? 

> The ranges are converted into array and then it is passed to the array overload.

Note we want the implementation that _doesn't_ create a temporary array.

Reply:
> What is the intention with this overload?

@ben-albrecht as you suggested to add an overload for handling ranges as well, I have added this overload. As we can also have strided ranges just like  strided arrays, to handle those cases and in general I have converted the range into an array and passed it to the array overload.

Reply:
> Note we want the implementation that _doesn't_ create a temporary array.

We are handling arrays in the overloads only. Operations in the `_choice` function are peformed on a domain only. Since we have to take care of the strided cases and we are returning indices now we have to deal with the `low` value and the `stride` of an array/range.